File: Introductory Problems/Two Sets/code.js
const fs = require('fs');
const input = fs.readFileSync(0, 'utf-8').trim().split('\n');
const n = parseInt(input[0], 10);

let sum = n * (n+1)/2;
if (sum&1) {
    console.log("NO");
    process.exit(0);
}
sum = sum /2;
let s1 = 0;
const set1 = []
const set2 = []

for(let i = n; i>=1; i--) {
    if (s1 + i <= sum) {
        set1.push(i);
        s1 += i;
    } else {
        set2.push(i);
    }
}
console.log("YES")
console.log(set1.length);
console.log(set1.join(' '))
console.log(set2.length);
console.log(set2.join(' '))

File: Introductory Problems/Two Sets/code.py
n = int(input())
s = n * (n+1) // 2
if s&1:
    print("NO")
    exit(0)
s = s//2

set1 = []
set2 = []

s1 = 0
for i in reversed(range(1, n+1)):
    if s1 + i <= s:
        set1.append(i)
        s1 += i
    else:
        set2.append(i)
        
print("YES")
print(len(set1))
print(*set1)
print(len(set2))
print(*set2)
